Client/mods/deathmatch/logic/CClientModel.cpp

```diff
@@ -53,20 +73,70 @@ bool CClientModel::Deallocate(void)
     // ModelInfo must be valid
     if (pModelInfo->IsValid())
     {
-        if (m_eModelType == CCLIENTMODELPED)
+        if (m_eModelType == eClientModelType::PED)
```

Comment:
I think a `switch` should be used here.
Also, maybe consider adding a lambda(obv. inside the function, as it will be inlined) for calling the event, instead of copy-pasting the code.

Reply:
Idk, but switch don't work here

Reply:
Here you go, the refactored code:
```cpp
bool CClientModel::Deallocate(void)
{
    if (!m_bAllocatedByUs)
        return false;

    CModelInfo* pModelInfo = g_pGame->GetModelInfo(m_iModelID, true);
    if (!pModelInfo) // ModelInfo must be valid
        return false;

    auto unloadModelsAndCallEvents = [&](auto iterBegin, auto iterEnd, auto& element, unsigned short usParentID, auto setElementModelLambda) {
        for (auto iter = iterBegin; iter != iterEnd; iter++)
        {
            auto& element = **iter;

            if (element.GetModel() != m_iModelID)
                continue;

            if (element.IsStreamedIn())
                element.StreamOutForABit();

            setElementModelLambda(element);

            CLuaArguments Arguments;
            Arguments.PushNumber(m_iModelID);
            Arguments.PushNumber(usParentID);
            element.CallEvent("onClientElementModelChange", Arguments, true);
        }
    };

    switch (m_eModelType)
    {
    case eClientModelType::PED:
    {
        // If some ped is using this ID, change him to CJ
        CClientPedManager*                       pPedManager = g_pClientGame->GetManager()->GetPedManager();

        unloadModelsAndCallEvents(pPedManager->IterBegin(), pPedManager->IterEnd(), 0, 
            [](auto& element) { element.SetModel(0); });

        break;
    }
    case eClientModelType::OBJECT:
    {
        const auto& objects pObjectManager = g_pClientGame->GetManager()->GetObjectManager()->GetObjects();
        unsigned short usParentID = g_pGame->GetModelInfo(m_iModelID)->GetParentID();

        unloadModelAndCallEvent(objects->begin(), objects->end(), usParentID, 
            [=](auto& element) { element.SetModel(usParentID); });

        break;
    }
    case eClientModelType::VEHICLE:
    {
        CClientVehicleManager* pVehicleManager = g_pClientGame->GetManager()->GetVehicleManager();
        unsigned short         usParentID = g_pGame->GetModelInfo(m_iModelID)->GetParentID();

        unloadModelAndCallEvent(pVehicleManager->IterBegin(), pVehicleManager->IterEnd(), usParentID, 
            [=](auto& element) { element.SetModelBlocking(usParentID, 255, 255); });

        break;
    }

    // Restore DFF/TXD
    g_pClientGame->GetManager()->GetDFFManager()->RestoreModel(m_iModelID);

    // Restore COL (for non ped models)
    if (m_eModelType != eClientModelType::PED)
        g_pClientGame->GetManager()->GetColModelManager()->RestoreModel(m_iModelID);

    pModelInfo->DeallocateModel();

    this->SetParentResource(nullptr);

    return true;
}
```

Reply:
This code don't work for peds

Reply:
Thanks you!